Fix-User-Guide.md

```diff
@@ -0,0 +1,252 @@
+elsif exists("warning")
+  set_field("is_valid", "yes")
+  log("warning")
+else
+  set_field("is_valid", "yes")
+end
+
+# Binds - Loops
+
+do list(path: "foo", "var": "$i")
+  add_field("$i.bar", "baz")
+end
+
+# Selector
+if exists("error")
+   reject()
+end
+
+```
+
+**Functions** are used to add, change, remove or otherwise manipulate elements.
+
+**Conditionals** are used to control the processing of function so that they are not process with every workflow but only under certain conditions.
+
+**Selectors** can be used as hghlevel filter to filter the records you want.
+
+**Binds** are wrappers for one or more fixes. They give extra control functionality for fixes such as loops.
+All binds have the same syntax:
+
+```PERL
+do Bind(params,…)
+   fix(..)
+   fix(..)
+end
+```
+
+Find here a [list of all function, selectors, binds and conditionals](/Fix-function-and-Cookbook.md).
+
+
+## Addressing Pieces of Data - of: FIX-Path and the record structure in FIX
+
+Internally FIX knows arrays, objects/hashes and simple elements. How a format is translated is depending on the `decode-...` command in the MF Workflow. Only one thing is specific to the fix, as in Catmandu a repeated field is translated into a list depending on the real input data of the single record and elements with the suffix `[]` are interpreted as arrays.
+
+Since function manipulate, add or remove elements in a record, it is essential to understand the way on can adress source or target elements.
+
+e.g.:
+```PERL
+copy_field("<sourceField>", "<targetField>")
+```
+
+To adress the source or target element here, you need to provide the path to the element.
+Metafacture Fix is using a path-syntax that is JSON Path like but not identical. It also uses the dot notation but there are some differences with the path structure of arrays and repeated fields. Especially when working with JSON, YAML or records repeated fields.
+
+```
+a : simpleField
+b : c : objectField1
+   d : objectField2
+   e : objectField3
+f : repeatedField1
+f : repeatedField2
+f : repeatedField3
+g : - listElement1
+   - listElement2
+   - listElement3
+h : - i : listObjectElement1.1
+      j : listObjectElement1.2
+    - i : listObjectElement2.1
+      j : listObjectElement2.2
+k : l : m : o : deepNestedField
+```
+
+The path for a simple string-element is adressed by stating the element name: `a`
+For the fields with deeper structure you add a dot ‘.’. The path for elements in nested objects is stated by: `b.c` or `k.l.m.o`
+
+In an data set an element sometimes an element can have multiple instances. Different data models solve this possibility differently. XML-Records can have all elements multiple times, element repition is possible and in many schemas (partly) allowed. Repeatable elements also exist in JSON and YAML but are unusual.
+
+To point to a specific element you state the index number. To adress the value `repeatedField2` the path would be `f.2`` since the repeated field is handled as list.
```

Comment:
```suggestion
To point to a specific element you state the index number. To adress the value `repeatedField2` the path would be `f.2`` since the repeated field is handled as a list.
```